### PR TITLE
Add assignCreditToUploader option when pushing

### PR DESCRIPTION
**Pull request by aeng**
https://bugzilla.redhat.com/show_bug.cgi?id=1184299

See also:
Server: https://github.com/zanata/zanata-server/pull/678
Client: https://github.com/zanata/zanata-client/pull/46

*DO NOT MERGE UNTIL Zanata-Server implementation is done.*

### CODE REVIEW

File: zanata-common-api/src/main/java/org/zanata/rest/dto/resource/TextFlowTarget.java [hunk 1]
```diff
@@ -50,6 +50,14 @@ public Person getTranslator() {
         return translator;
     }
 
+    @Deprecated
+    /**
+     * Zanata server will ignore this field if set,
```
Comment (davidmason) [inline zanata-common-api/src/main/java/org/zanata/rest/dto/resource/TextFlowTarget.java]: I did not remember this being part of the criteria we decided, so I looked for the criteria and could not find any. Did we really decide to make the translator field not work?

@aeng What criteria are you using to implement this?
@djansen-redhat What criteria will you be using to verify this?

File: zanata-common-api/src/main/java/org/zanata/rest/service/AsynchronousProcessResource.java [hunk 2]
```diff
@@ -167,6 +167,9 @@ public ProcessStatus startSourceDocCreationOrUpdate(
      *            Auto will check the history of your translations and will not
      *            overwrite any translations for which it detects a previous
      *            value is being pushed.
+     * @param isMyTranslations
+     *            Uploader will take credit of all uploaded translations as
+     *            their own.
```
Comment (davidmason) [inline zanata-common-api/src/main/java/org/zanata/rest/service/AsynchronousProcessResource.java]: Recommend wording as "The translator field for all uploaded translations will be set to the user who performs the upload."

File: zanata-common-api/src/main/java/org/zanata/rest/dto/resource/TextFlowTarget.java (hunk 1 above)
Comment (aeng) [inline zanata-common-api/src/main/java/org/zanata/rest/dto/resource/TextFlowTarget.java]: We never used this field at all. To simplify implementation on server, this should be deprecated.

Comment (davidmason) [inline zanata-common-api/src/main/java/org/zanata/rest/dto/resource/TextFlowTarget.java]: Cool, if we never used it then it is fine to deprecate.

**Comment by davidmason** [comment]
:+1:

**Comment by aeng** [comment]
Merged into integration/master. https://github.com/zanata/zanata-api/commit/86e0008c9761100dc7f19f7e9585b596fd21383a

**Comment by aeng** [comment]
Merged into integration/master. https://github.com/zanata/zanata-api/commit/cd7548d739ada77723ad0b50af3ff7e9a808b426

**Comment by aeng** [comment]
Merged into integration/master. https://github.com/zanata/zanata-api/commit/bb9f2c73d60c6f92719fd1c4489635801f0cb3f0

**Pull request closed without merging**